samza-core/src/main/java/org/apache/samza/operators/impl/OperatorImpl.java

```diff
@@ -258,7 +258,7 @@ public final void aggregateEndOfStream(EndOfStreamMessage eos, SystemStreamParti
    * @param collector message collector
    * @param coordinator task coordinator
    */
-  private final void onEndOfStream(MessageCollector collector, TaskCoordinator coordinator) {
+  private void onEndOfStream(MessageCollector collector, TaskCoordinator coordinator) {
```

Comment:
Why do we remove the final declarative here? I thought that the reason Xinyu marked this as final is no operator should override this, since it is the system-controlled behavior.

Reply:
Private methods are final anyway, so was redundant. But I'll add it back so that this doesn't get changed on a future refactor.